patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -2711,6 +2909,81 @@
  			int num = genRand.Next(4);
  			int num2 = 0;
  			int num3 = 0;
+@@ -50715,6 +_,24 @@
+ 					if (CountNearBlocksTypes(num11, num12, 2, 1, 27) > 0)
+ 						continue;
+ 
++					int preHookType = Main.tile[num11, num12].type;
++					if (!TileLoader.Convert(num11, num12, 1)) {
++						if (preHookType == Main.tile[num11, num12].type)
++							continue;
++
++						if (genRand.Next(2) == 0)
++							flag2 = true;
++
++						continue;
++					}
```

Comment:
`bool convertTile = TileLoader.Convert(...)`

later, after the `preHookType`

`if (!convertTile) continue;`

Also, did you want to add the `HasTile` guards here too?